src/viur/core/request.py

```diff
@@ -568,73 +570,84 @@ def findAndCall(self, path: str, *args, **kwargs) -> None:
             raise errors.NotFound(
                 f"""The path {utils.escapeString("/".join(self.path_list[:idx]))} could not be found""")
 
-        if (not callable(caller) or ((not "exposed" in dir(caller) or not caller.exposed)) and (
-            not "internalExposed" in dir(caller) or not caller.internalExposed or not self.internalRequest)):
-            if "index" in caller \
-                and (callable(caller["index"]) \
-                     and ("exposed" in dir(caller["index"]) and caller["index"].exposed) \
-                     or ("internalExposed" in dir(
-                    caller["index"]) and caller["index"].internalExposed and self.internalRequest)):
-                caller = caller["index"]
+        if not isinstance(caller, Method):
+            # try to find "index" function
+            if (index := caller.get("index")) and isinstance(index, Method):
+                caller = index
             else:
                 raise errors.MethodNotAllowed()
-        # Check for forceSSL flag
+
+        # Check for internal exposed
+        if caller.exposed is False and not self.internalRequest:
```

Comment:
`None` should not be a problem, shouldn't it?

```suggestion
        if not caller.exposed and not self.internalRequest:
```

Reply:
This line is correct.

exposed is
- True: the function is generally exposed
- False: the function is internally exposed
- None: the function is not exposed